packages/astro/src/vite-plugin-astro/index.ts

```diff
@@ -103,35 +102,22 @@ export default function astro({ settings, logging }: AstroPluginOptions): vite.P
 			if (!query.astro) {
 				return null;
 			}
-			let filename = parsedId.filename;
-			// For CSS / hoisted scripts we need to load the source ourselves.
-			// It should be in the compilation cache at this point.
-			let raw = await this.resolve(filename, undefined);
-			if (!raw) {
-				return null;
-			}
-
-			let source = getCachedSource(config, raw.id);
-			if (!source) {
+			// For CSS / hoisted scripts, the main Astro module should already be cached
+			const filename = normalizeFilename(parsedId.filename, config);
+			const compileResult = getCachedCompileResult(config, filename);
+			if (!compileResult) {
 				return null;
 			}
```

Comment:
I found `this.resolve` and `normalizeFilename` seems to guard the same thing conceptually, so I opt for `normalizeFilename` as it's done by other plugins too.